frappe/public/js/frappe/form/controls/attach.js

```diff
@@ -79,10 +79,12 @@ frappe.ui.form.ControlAttach = frappe.ui.form.ControlData.extend({
 		this.value = value;
 		if(this.value) {
 			this.$input.toggle(false);
-			if(this.value.indexOf(",")!==-1) {
-				var parts = this.value.split(",");
-				var filename = parts[0];
-				dataurl = parts[1];
+			// value can also be using this format: FILENAME,DATA_URL
+			// Important: We have to be careful because normal filenames may also contain ","
+			let two_part_matches = this.value.match(/^([^:]+),(.+):(.+)$/);
+			if (two_part_matches !== null) {
+				let filename = two_part_matches[1];
```

Comment:
Please declare filename outside the `if` block so that it can be accessed outside it.  Variables declared using `let` cannot be accessed outside the block in which they are declared. This is also why Sider is failing.

```suggestion
			let two_part_matches = this.value.match(/^([^:]+),(.+):(.+)$/);
		        let filename;
			if (two_part_matches !== null) {
				filename = two_part_matches[1];
```

Reply:
@sagarvora Can you re-clarify about this? These variables are specifically never (and shouldn't be) used outside of these blocks.

If there are other functions who need these variables, can you point which ones? Because these functions will need improvements are well, as we should resort to global variables only when there is no better alternative.

Reply:
@ceefour `filename` is used outside the `if` block where it has been declared, here:
https://github.com/ceefour/frappe/blob/09886799e8badea0bb55d2024ab292a13e419632/frappe/public/js/frappe/form/controls/attach.js#L90

Unlike `var`, variables declared using `let` are scoped to the immediate enclosing block.
You will have to declare the variable outside the `if` block as @sagarvora has suggested.